src/plat/bcm2711/overlay-rpi4.dts

```diff
@@ -21,4 +21,70 @@
 		/* This is configurable in the Pi's config.txt, but we use 128MiB of RAM by default. */
 		reg = <0x00 0x00000000 0x08000000>;
 	};
+
+	soc {
```

Comment:
Ok.  So after quickly looking through Section 1.2 (Address Map) of the BCM2711 ARM Peripherals document (https://datasheets.raspberrypi.com/bcm2711/bcm2711-peripherals.pdf), I think I understand what you mean better.  The addresses  in the device tree base file correspond to a legacy view that is still valid for DMA addresses but would be invalid for Arm core MMIO addresses.  In BCM2711, accessing the old addresses from the ARM core would actually be accessing regions of the SDRAM right?  In this case I agree that it does make sense to correct the addresses to the ARM view as you suggest in this PR.

Reply:
Yes, that is pretty much the problem with these Raspberry Pi boards. In this PR, we only look at the UART peripherals and the GPIO (for pin configuration) since these are the peripherals affected by our use case. But generally, this situation of rewriting the base address would also need to be done for other peripherals. I suppose it is sufficient for this PR to only modify the UARTs and GPIO right?

Reply:
I was talking with @axel-h already about it, but the question is if all this information that is introduced with this PR should be put into the overlay-rpi4.dts file. My initial understanding was that anything that is different from the Linux DTS file should be put into the overlay file. This might very easily inflate the overlay file. But what about the pin configuration? Since UARTs 2-5 are deactivated by default, there is no pin configuration taking place in rpi4.dts. Would it make sense to move the pin configuration to the rpi4.dts file and only do the address translation in the overlay-rpi4.dts file? I suppose this would keep the overlay-rpi4.dts file slim.

Reply:
You can have more than one overlay file, perhaps keep the Linux one unmodified as much as possible, add one for pin control and one for address translations if you find it too cluttered if it's in one file?

Reply:
I know, but that is also the reason why I am not sure what the best solution is. If the Linux dts file should not be modified, then maybe we need two additional files as you proposed:

- overlay-rpi4-address-mapping.dts: maps for all peripherals the address space from VC to ARM, I would add a comment that additional peripherals should be added. But for our use case, we currently only need GPIO and UARTs

- overlay-rpi4-uart.dts: This will contain the pin configuration as well as the activation of the peripheral itself

Does this sound good to you?

Reply:
I added two dts files in my third commit. Which of the three commits/solutions do you like best?

Reply:
> But what about the pin configuration?

Is this pin configuration reflecting what the configuration is when seL4 starts?

Reply:
Thank you for this question. It seems like we need to do two things to get proper UART support:
- Set the respective pins to ALT4. For this it is sufficient to use the gpio driver in libplatsupport. There is no need to set the pins in the dts file.
- Have the correct peripheral address mapping from VC to ARM in a dts file.

I think the address mapping is all we need to do the DTS file.